### PR TITLE
Removing keystore and adding pem SSL support

### PR DESCRIPTION
**Pull request by RobHooper**
Addressing issue #387

When we deploy we also need to action:
1. Remove the following files from the server
>  /opt/pkcs-password
>  /opt/pem-to-keystore.sh
>  /opt/pem-to-keystore-wrapper.sh
>  /etc/sudoers.d/90-pem-to-keystore 
2. Remove ``key_pass`` from pillar/private ``elasticsearch:plugins:readonlyrest:key_pass``

### CODE REVIEW

File: salt/elasticsearch/files/config/readonlyrest-docs.yml [hunk 1]
```diff
@@ -2,9 +2,8 @@ readonlyrest:
   # https://github.com/beshu-tech/readonlyrest-docs/blob/master/kibana.md#loading-settings-order-of-precedence
   force_load_from_file: True
   ssl:
-    keystore_file: keystore.jks
-    keystore_pass: {{ pillar.elasticsearch.plugins.readonlyrest.key_pass }}
-    key_pass: {{ pillar.elasticsearch.plugins.readonlyrest.key_pass }}
+    server_certificate_key_file: /etc/apache2/md/domains/standard.open-contracting.org/privkey.pem
```
Comment (RobHooper) [inline salt/elasticsearch/files/config/readonlyrest-docs.yml:5]: I don't think this should be hard-coded, perhaps a variable like: ``elasticsearch:plugins:readonlyrest:privkey``

Comment (jpmckinney) [reply salt/elasticsearch/files/config/readonlyrest-docs.yml:5]: Sounds good to me.

File: salt/elasticsearch/plugins/readonlyrest.sls [hunk 2]
```diff
@@ -12,45 +12,27 @@ readonlyrest-install:
...
-      - file: /opt/pkcs-password
-
-/opt/pem-to-keystore-wrapper.sh:
-  file.managed:
-    - name: /opt/pem-to-keystore-wrapper.sh
-    - source: salt://elasticsearch/files/pem-to-keystore-wrapper.sh
+    - name: /opt/restart-elasticsearch.sh
+    - source: salt://elasticsearch/files/restart-elasticsearch.sh
     - user: root
     - group: root
```
Comment (jpmckinney) [inline salt/elasticsearch/plugins/readonlyrest.sls:20]: These are from before, but we can probably remove the user/group root lines.

File: salt/elasticsearch/plugins/readonlyrest.sls [hunk 3]
```diff
@@ -12,45 +12,27 @@ readonlyrest-install:
...
     - mode: 755
```
Comment (jpmckinney) [inline salt/elasticsearch/plugins/readonlyrest.sls:21]: Change to 750 if only run via sudo/root?

File: salt/elasticsearch/files/config/readonlyrest-docs.yml (hunk 1 above)
Comment (RobHooper) [reply salt/elasticsearch/files/config/readonlyrest-docs.yml:5]: Done ✔

File: salt/elasticsearch/plugins/readonlyrest.sls (hunk 3 above)
Comment (RobHooper) [reply salt/elasticsearch/plugins/readonlyrest.sls:21]: Done ✔

Comment (RobHooper) [reply salt/elasticsearch/plugins/readonlyrest.sls:21]: This needs to be executable (and not editable) for the ``www-data`` user which 750 will block.

Alternatively we could set the group to ``www-data`` however this is quite specific to the docs server and won't scale as easily if other users need access.

File: salt/elasticsearch/plugins/readonlyrest.sls (hunk 2 above)
Comment (RobHooper) [reply salt/elasticsearch/plugins/readonlyrest.sls:20]: Done ✔

File: salt/elasticsearch/plugins/readonlyrest.sls (hunk 3 above)
Comment (jpmckinney) [reply salt/elasticsearch/plugins/readonlyrest.sls:19]: Ah, right, this file is run by www-data, and the command within this file is sudo 👍

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**